components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/BPMNDataPublisher.java

```diff
@@ -236,4 +340,182 @@ private String getProcessStreamId() {
+            //get a list of names of variables which are configured for analytics from registry for that process, if
+            // not already taken from registry
+            String[][] configedProcessVariables;
+            if (processVariablesMap.get(processDefinitionId) == null) {
+                JsonObject kpiConfig = getKPIConfiguration(processDefinitionId);
+                //do not publish the KPI event if DAS configurations are not done by the PC
+                if (kpiConfig == null) {
+                    return;
+                }
+
+                /* Keeps configed process variabe data as a JSON. Example value:
+                [{"isAnalyzeData":false,"name":"size","isDrillDownData":false,"type":"int"},
+                {"isAnalyzeData":false,"name":"status","isDrillDownData":false,"type":"string"},
+                {"isAnalyzeData":false,"name":"pizzaTopping","isDrillDownData":false,"type":"string"},
+                {"isAnalyzeData":false,"name":"amount","isDrillDownData":false,"type":"int"},
+                {"isAnalyzeData":"false","name":"processInstanceId","isDrillDownData":"false","type":"string"}]
+                 */
+                JsonArray configedProcVarsJson = kpiConfig
+                        .getAsJsonArray(AnalyticsPublisherConstants.PROCESS_VARIABLES_JSON_ENTRY_NAME);
+
+                int variableCount = configedProcVarsJson.size();
+                configedProcessVariables = new String[variableCount][2];
+
+                for (int i = 0; i < variableCount; i++) {
+                    configedProcessVariables[i][0] = ((JsonObject) configedProcVarsJson.get(i)).get("name")
+                            .getAsString();
+                    configedProcessVariables[i][1] = ((JsonObject) configedProcVarsJson.get(i)).get("type")
+                            .getAsString();
+                }
+
+                processVariablesMap.put(processDefinitionId, configedProcessVariables);
+                eventStreamId = kpiConfig.get("eventStreamId").getAsString();
+                kpiStreamIdMap.put(processDefinitionId, eventStreamId);
+
+            } else { //if the process variables are already taken, get them from the Map
+                configedProcessVariables = processVariablesMap.get(processDefinitionId);
+                eventStreamId = kpiStreamIdMap.get(processDefinitionId);
+            }
+
+            Map<String, VariableInstance> variableInstances = ((ExecutionEntity) processInstance)
+                    .getVariableInstances();
+            payload = new Object[configedProcessVariables.length];
+
+            for (int i = 0; i < configedProcessVariables.length - 1; i++) {
+                String varName = configedProcessVariables[i][0];
+                String varType = configedProcessVariables[i][1];
+
+                Object varValue = variableInstances.get(varName).getValue();
+
+                switch (varType) {
+                case "int":
+                    if (varValue == null) {
```

Comment:
Thanks @cnapagoda for the comment. If the KPI data publishing is configured by Process Center in BPS,those variable details are stored in BPS registry. But if no org.activiti.engine.impl.persistence.entity.VariableInstance (related to that variable) is created while BPMN process execution, there is no such variable value for that variable. If so, varValue is null and so we attach default values for those variables in the payload, also indicating that they have no value using a 'valueAvailability' flag. If the varValue is not null, we can attach actual value value to payload.